compiler-plugin/src/main/java/io/ballerina/stdlib/log/compiler/staticcodeanalyzer/LogStatementAnalyzer.java

```diff
@@ -0,0 +1,147 @@
+                            if (expression instanceof SimpleNameReferenceNode simpleNameReferenceNode) {
+                                checkConfigurableQualifier(simpleNameReferenceNode, context, expressionStatementNode);
+                            } else if (expression instanceof TemplateExpressionNode templateExpressionNode) {
+                                templateExpressionNode.content().forEach(content -> {
+                                    if (content instanceof InterpolationNode interpolationNode) {
+                                        interpolationNode.childEntries().forEach(interpolationChild -> {
+                                            Node interpolationExpression = interpolationChild.node().orElse(null);
+                                            if (interpolationExpression instanceof SimpleNameReferenceNode
+                                                    simpleNameReferenceNode) {
+                                                checkConfigurableQualifier(simpleNameReferenceNode, context,
+                                                        expressionStatementNode);
+                                            }
+                                        });
+                                    }
+                                });
+                            } else if (expression instanceof BinaryExpressionNode binaryExpressionNode) {
+                                binaryExpressionNode.childEntries().forEach(childEntry -> {
+                                    Node childNode = childEntry.node().orElse(null);
+                                    if (childNode instanceof SimpleNameReferenceNode simpleNameReferenceNode) {
+                                        checkConfigurableQualifier(simpleNameReferenceNode, context,
+                                                expressionStatementNode);
+                                    }
+                                });
+                            }
+                        });
+                    } else if (node instanceof NamedArgumentNode namedArgumentNode) {
+                        checkConfigurableQualifier(namedArgumentNode.expression(), context, expressionStatementNode);
+                    }
+                }
+            }
+        }
+    }
+
+    private void checkConfigurableQualifier(ExpressionNode argumentNode, SyntaxNodeAnalysisContext context,
+                                            ExpressionStatementNode expressionStatementNode) {
+        semanticModels.forEach(semanticModel -> {
+            Symbol symbol = semanticModel.symbol(argumentNode).orElse(null);
+            if (symbol instanceof VariableSymbol variableSymbol) {
+                variableSymbol.qualifiers().stream().filter(qualifier -> qualifier
+                        .toString().equals(CONFIGURABLE_QUALIFIER)).forEach(qualifier -> {
+                    this.reporter.reportIssue(getDocument(context),
+                            expressionStatementNode.location(),
```

Comment:
Shall we give the error at the argument nodes instead? Otherwise, when a log statement logs multiple configurables, they'll all be all at the same location, right?

Reply:
Can probably add a test like that too.

Reply:
Fixed in [81d0c7d](https://github.com/ballerina-platform/module-ballerina-log/pull/1175/commits/81d0c7d8d9fa4aa8a0814c2882a1a506e15dd47c)